### PR TITLE
RavenDB-22937: Adjusting Corax's search to match Lucene's search method & tests for search behavior

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22937

### Additional description

Tests for new search behavior when a wildcard occurs in the search phrase.

Corax:
- Introduced corax new search method to match Lucene's search method
- Corax: backward compatibility for older indexes

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [x] New feature

### How risky is the change?

- [ ] Low 
- [x] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [x] Ensured. New behavior is available for new indexes.
- [ ] Breaking change
- [] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by arekpalinski — DISMISSED** [review]

File: test/SlowTests/Issues/RavenDB_22937.cs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+public class RavenDB_22937 : RavenTestBase
+{
+    public RavenDB_22937(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [RavenFact(RavenTestCategory.Querying)]
+    public void StaticIndexStandardAnalyzerCanSearchForWildcardsInSearch()
+    {
+        using var store = GetDocumentStore(Options.ForSearchEngine(RavenSearchEngineMode.Lucene));
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB_22937.cs]: Can we check it on Corax as well?

**Comment by arekpalinski** [comment]
test this in debug please

**Comment by arekpalinski** [comment]
Can you please elaborate more on changes made in Corax? It can be added as comments of relevant changes. Is is safe to introduce it without bumping index version?

**Comment by arekpalinski** [comment]
Can you please update issue title and description (after adjusting the behavior for Corax indexes as well)?

**Comment by arekpalinski** [comment]
Is it sill draft?

**Review by arekpalinski — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Pull request merged**